File: 2020/Day3/Day3-1.py
f = open("Day3.txt", "r")
arr = f.readlines()
x = 0
count = 0
length = len(arr[0])-1

for i in range(len(arr)):
    count += 1 if arr[i][x] == "#" else 0
    x = (x+3)%length
print (count)

File: 2020/Day3/Day3-1.c
#include <stdio.h>

int main () {
    //variables
    FILE *fp = fopen("Day3.txt", "r");
    char arr[500][32];
    int lines=0;
    int left=0;
    int k=0;
    int count=0;
    
    //take in input
    while (lines<323 && fscanf(fp, "%s", arr[lines])) {
        lines += 1;
    }

    //calculate answer
    for (k=0; k<lines; k++) {
        if (arr[k][left]=='#') 
            count += 1;
        left+=3;
        left%=31;
    }
    printf ("%d\n", count);
    return 0;
}
